### ISSUE TITLE
UI elements or callback

### ISSUE THREAD

**Issue by marcotrumpet** [eopeter/flutter_mapbox_navigation#229]
Hi, is it possible to customise or show/hide specific elements (like report problem screen or end navigation screen)? Or maybe is there some callback to intercept those events? Right know are just buttons that does nothing (if I rate the navigation where those infos go?).

Would be interesting also to know how to translate text into those screens

Thanks

**Comment by dschandrasekara** [comment]
@marcotrumpet similar kind of [issue](https://github.com/eopeter/flutter_mapbox_navigation/issues/230), i also have.  i think, if we can get mapbox event , when navigation is closed, it is really helpful to us update the our UIs.

**Comment by marcotrumpet** [comment]
> @marcotrumpet similar kind of [issue](https://github.com/eopeter/flutter_mapbox_navigation/issues/230), i also have. i think, if we can get mapbox event , when navigation is closed, it is really helpful to us update the our UIs.

I'm working on a pull request to add 2 parameters into options. Those will be boolean to show or not the feedback button and the end of navigation screen on iOS.
Then I'll be able to catch the end of navigation and handle it in my Flutter app

**Comment by marcotrumpet** [comment]
That's the PR https://github.com/eopeter/flutter_mapbox_navigation/pull/231 
After that I'm able to catch the end of navigation using 
```dart
  Future<void> _onEmbeddedRouteEvent(RouteEvent e) async {
    switch (e.eventType) {
      case MapBoxEvent.progress_change:
        final progressEvent = e.data as RouteProgressEvent;
        if (progressEvent.arrived ?? false) {
          await _controller.finishNavigation();
        }
    }
  }
```

**Comment by dschandrasekara** [comment]
@marcotrumpet wow nice, can I track close event ?
![818A726E-253D-42FA-98EC-6C95BBE3923A](https://github.com/eopeter/flutter_mapbox_navigation/assets/56452781/63e2b7fc-9e47-40cc-a2c2-584429b17393)

Btw, when I click this close button, it will be pop up navigation screen and showing my previous screen and it is I’m expecting behavior and that’s working only for iOS. In android app, it will be redirected to another MapView and showing current location only and I don’t have any close option for that screen and can’t go to my previous screen, then user can’t access other features of my app, but it is only for android side
Any idea about that?
Note - I didn’t use MapBoxNavigationViewController and used only MapBoxNavigation.instance.startNavigation() method only

**Comment by marcotrumpet** [comment]
If your problem is on Android it could be because none of navigation_finished or navigation_cancelled is ever called 
<img width="552" alt="Screenshot 2023-05-20 alle 18 35 38" src="https://github.com/eopeter/flutter_mapbox_navigation/assets/35219231/d29d8213-0346-4b58-a61f-b9ab37a3ca07">

**Comment by dschandrasekara** [comment]
@marcotrumpet yeah, if we can track that events, it’s really helpful to us for building our logic, but that events are not fired iOS app also, but navigation screen will be closed and move to my previous screen, two different behavior for android and iOS apps

**Comment by dschandrasekara** [comment]
> @marcotrumpet yeah, if we can track that events, it’s really helpful to us for building our logic, but that events are not fired iOS app also, but navigation screen will be closed and move to my previous screen, two different behavior for android and iOS apps



> If your problem is on Android it could be because none of navigation_finished or navigation_cancelled is ever called
> <img alt="Screenshot 2023-05-20 alle 18 35 38" width="552" src="https://user-images.githubusercontent.com/35219231/239699027-d29d8213-0346-4b58-a61f-b9ab37a3ca07.png">

@marcotrumpet is it included in any future PRs?

**Comment by marcotrumpet** [comment]
> > @marcotrumpet yeah, if we can track that events, it’s really helpful to us for building our logic, but that events are not fired iOS app also, but navigation screen will be closed and move to my previous screen, two different behavior for android and iOS apps
> 
> 
> 
> 
> 
> 
> 
> > If your problem is on Android it could be because none of navigation_finished or navigation_cancelled is ever called
> 
> > <img alt="Screenshot 2023-05-20 alle 18 35 38" width="552" src="https://user-images.githubusercontent.com/35219231/239699027-d29d8213-0346-4b58-a61f-b9ab37a3ca07.png">
> 
> 
> 
> @marcotrumpet is it included in any future PRs?

I'm working on Android side, iOS seems fine.. I'll push both together if I will able to fix it

**Comment by dschandrasekara** [comment]
@marcotrumpet thanks a lot mate !!!, yeah , you are 100% correct and it’s working fine for iOS and issue is only for iOS side, we already deployed to iOS app to apple play store and waiting to this fix for android to google play store submission, I really appreciate your help to fix this issue 👍

**Comment by marcotrumpet** [comment]
https://github.com/eopeter/flutter_mapbox_navigation/pull/235

**Comment by eopeter** [comment]
Will need to look into how the specific elements can be hidden or at least provide a way to replace with yours

**Comment by marcotrumpet** [comment]
> @marcotrumpet thanks a lot mate !!!, yeah , you are 100% correct and it’s working fine for iOS and issue is only for Android side, we already deployed to iOS app to apple play store and waiting to this fix for android to google play store submission, I really appreciate your help to fix this issue 👍

Since the owner of this package doesn't want to merge feel free to refer my repo in your pubspec. Enjoy the fix

**Comment by dschandrasekara** [comment]
@marcotrumpet @eopeter after i installed [0.1.3](https://pub.dev/packages/flutter_mapbox_navigation/versions/0.1.3) and it is fired successfully navigation cancelled event in android side, when user click the close button of navigation view. I tracked that event, and just call MapBoxNavigation.instance.finishNavigation() method. So then it redirected to my previous screen. The issue has been resolved now, Thank you very much guys !!!!
